File: zodiac.js
let zodiac = "鼠牛虎兔龙蛇马羊猴鸡狗猪";


function zodiac(year) {
    let zodiac = "鼠牛虎兔龙蛇马羊猴鸡狗猪";
    return zodiac[(year - 4) % 12];
}


class Zodiac {
    constructor() {
        this.zodiac = "鼠牛虎兔龙蛇马羊猴鸡狗猪";
    }

    zodiac(year) {
        let zodiac = "鼠牛虎兔龙蛇马羊猴鸡狗猪";
        return zodiac[(year - 4) % 12];
    }
}

File: zodiac.r
zodiac <- c("鼠", "牛", "虎", "兔", "龙", "蛇", "马", "羊", "猴", "鸡", "狗", "猪")
branch <- c("子", "丑", "寅", "卯", "辰", "巳", "午", "未", "申", "酉", "戌", "亥")
branchzodiac <- function(year) {
  return(paste0(branch[(year-3)%%12], zodiac[(year-3)%%12]))
}
